### ISSUE TITLE
filtering on top of card built with custom builder that calls a table created from query builder

### ISSUE THREAD

**Issue by manoloah** [metabase/metabase#9012]
### Feature requests and proposals
When I create a card with a query builder and then I use the custom builder over the created table for some reason I cannot add a filter on top of the card. It would be awesome to enable this because, in my opinion, a best practice of creating dashboards is to use one main view or card (in case it gets modified you only modify the source card) and then generate charts on top of it so in case you have to change your code in the query builder you do it once and the changes gets reflected on the rest of the cards in the dashboard.

See example below of issue: 

1. I created a query using the query builder: 
![screen shot 2018-11-23 at 3 48 49 pm](https://user-images.githubusercontent.com/15304558/48961050-4da99800-ef37-11e8-92b6-3b4a0648b0d4.png)

2. Then I add this to my dashboard as my master card (or master view) 

![screen shot 2018-11-23 at 3 51 04 pm](https://user-images.githubusercontent.com/15304558/48961094-9b260500-ef37-11e8-99fa-fa43c72e5279.png)

3. Then I create a custom question from the table created as the master table: 
![screen shot 2018-11-23 at 3 51 42 pm](https://user-images.githubusercontent.com/15304558/48961110-b1cc5c00-ef37-11e8-9844-13fec8b50e98.png)

4. I add it to the same dashboard: 
![screen shot 2018-11-23 at 3 52 17 pm](https://user-images.githubusercontent.com/15304558/48961118-c872b300-ef37-11e8-8492-11d56c0035b6.png)

5. Then I try to filter both tables but the custom question has no filtering options which is the main request. That a custom question built from a question done with the query builder can also be filterable: 
![screen shot 2018-11-23 at 3 54 04 pm](https://user-images.githubusercontent.com/15304558/48961162-07086d80-ef38-11e8-8707-0656335afdef.png)

See: "No valid fields" this should let you filter with the user_user_id column.

**Comment by jlisyc** [comment]
Also bug related for 0.31.2 https://github.com/metabase/metabase/issues/9267

**Comment by flamber** [comment]
Hi @manoloah 
I was able to reproduce this on 0.31.2, but if I change the SQL query to use the Entity Key for the foreign table instead of using Foreign Key on the main table, then I can make it work.
```SQL
SELECT
    PEOPLE.ID,
    ORDERS.CREATED_AT,
    ORDERS.TOTAL
FROM ORDERS
LEFT JOIN PEOPLE ON PEOPLE.ID=ORDERS.USER_ID
WHERE ORDERS.QUANTITY > 1 [[AND PEOPLE.ID={{USER_ID}}]]
```

**Comment by DannyK85** [comment]
Found this ticket that looks like a similar issue/question I am having.
I have a native query (postgres) that uses a filter, [[where language = {{category}}]].
But when adding a filter to the dashboard I am only getting no valid fields.
Metabase version 0.31.2

Is there a way to use filters in the dashboard that link to a native sql question, not a direct database source?

**Comment by flamber** [comment]
Hi @DannyK85 
If you're using Field Filters, then your query should look like `[[where {{category}}]]`, since the referenced column is defined in the side panel. Check the updated documentation:
https://metabase.com/docs/latest/users-guide/13-sql-parameters.html#optional-clauses
Is there a specific reason why you're not using the latest 0.32.9 release?

**Comment by DannyK85** [comment]
I can't use field filters, since the field I want to filter consists of more values then from a single database. It's a combination of a couple.

So far the latest updates, 0.32.9, are draining Postgres database connections for some reason.
It's doing that for the last couple of versions.

**Comment by flamber** [comment]
@DannyK85 
The [upcoming](https://discourse.metabase.com/t/preview-upcoming-changes-to-the-gui-query-builder-updated/6551) version 0.33 will allow joins, so the need for SQL queries should be much less.
There's a couple of issues open about Postgres hanging connections - but without further details, it's difficult to point to a specific issue - maybe you're seeing issue 8679.

**Comment by flamber** [comment]
This is a duplicate of #9802, which is fixed in the upcoming 0.36.0
